File: app/src/main/java/com/android/FinalNotes/ui/MainActivity.java
package com.android.FinalNotes.ui;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.android.FinalNotes.R;
import com.android.FinalNotes.RouterHolder;
import com.android.FinalNotes.ui.auth.AuthFragment;
import com.google.android.material.bottomnavigation.BottomNavigationView;

public class MainActivity extends AppCompatActivity implements RouterHolder {

    private MainRouter router;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        router = new MainRouter(getSupportFragmentManager());

        if (savedInstanceState == null) {
            router.showAuth();
        }

        BottomNavigationView bottomNavigationView = findViewById(R.id.bottom_navigation);

        bottomNavigationView.setOnNavigationItemSelectedListener(item -> {
            if (item.getItemId() == R.id.action_notes) {
                router.showAuth();
            }

            if (item.getItemId() == R.id.action_info) {
                router.showInfo();
            }
            return true;
        });

        getSupportFragmentManager().setFragmentResultListener(AuthFragment.AUTH_RESULT, this, (requestKey, result) -> router.showNotes());
    }

    @Override
    public MainRouter getMainRouter() {
        return router;
    }
}

File: app/src/main/java/com/android/FinalNotes/ui/MainActivity.kt
package com.android.FinalNotes.ui

import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import com.android.FinalNotes.R
import com.android.FinalNotes.RouterHolder
import com.android.FinalNotes.ui.auth.AuthFragment
import com.google.android.material.bottomnavigation.BottomNavigationView

class MainActivity : AppCompatActivity(), RouterHolder {
    override var mainRouter: MainRouter? = null
        private set

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        mainRouter = MainRouter(supportFragmentManager)
        if (savedInstanceState == null) {
            mainRouter!!.showAuth()
        }
        val bottomNavigationView = findViewById<BottomNavigationView>(R.id.bottom_navigation)
        bottomNavigationView.setOnNavigationItemSelectedListener { item: MenuItem ->
            if (item.itemId == R.id.action_notes) {
                mainRouter!!.showAuth()
            }
            if (item.itemId == R.id.action_info) {
                mainRouter!!.showInfo()
            }
            true
        }
        supportFragmentManager.setFragmentResultListener(AuthFragment.AUTH_RESULT, this, { requestKey: String?, result: Bundle? -> mainRouter!!.showNotes() })
    }
}